src/subsplit_dag.hpp

```diff
@@ -60,6 +60,71 @@ class SubsplitDAG {
     SizeVector new_node_ids, new_edge_idxs, node_reindexer, edge_reindexer;
   };
 
+  // Nearest Neighbor Interchange Operations
+  class NNIOperation {
+   public:
+    NNIOperation(Bitset parent, Bitset child) : parent_(parent), child_(child){};
+
+    static int Compare(const NNIOperation &nni_a, const NNIOperation &nni_b) {
+      auto compare_parent = Bitset::Compare(nni_a.parent_, nni_b.parent_);
+      if (compare_parent != 0) {
+        return compare_parent;
+      }
+      auto compare_child = Bitset::Compare(nni_a.child_, nni_b.child_);
+      return compare_child;
+    };
+    int Compare(const NNIOperation &nni_b) {
+      const NNIOperation &nni_a = *this;
+      return Compare(nni_a, nni_b);
+    }
+
+    friend bool operator<(const NNIOperation &lhs, const NNIOperation &rhs) {
+      return Compare(lhs, rhs) < 0;
+    };
+    friend bool operator==(const NNIOperation &lhs, const NNIOperation &rhs) {
+      return Compare(lhs, rhs) == 0;
+    };
+
+    Bitset parent_;
+    Bitset child_;
+  };
+
+  class SetOfNNIs {
+   public:
+    SetOfNNIs() : set_(){};
+
+    friend bool operator==(const SetOfNNIs &lhs, const SetOfNNIs &rhs) {
+      return lhs.set_ == rhs.set_;
+    };
+
+    void insert(NNIOperation nni_op) { set_.insert(nni_op); };
```

Comment:
I realize that I did this lower-case-convention with Bitset, but in retrospect it seems like a bad idea. What do you think?